analyzers/src/SonarAnalyzer.Common/AnalysisContext/SonarAnalysisContextBase.cs

```diff
@@ -47,6 +47,9 @@ public abstract class SonarAnalysisContextBase<TContext> : SonarAnalysisContextB
     public SonarAnalysisContext AnalysisContext { get; }
     public TContext Context { get; }
 
+    private readonly ConcurrentDictionary<SyntaxTree, bool> shouldAnalyzeTreeCache = new();
+    private bool? shouldAnalyzeNullTreeCache = null;
```

Comment:
For now, I use this single bool to cache the result when the SyntaxTree is null as I cannot store `null` as a key value in the ConcurrentDictionary.